### PR TITLE
Migrate to scheduler API #71

### PR DESCRIPTION
**Pull request by ashklianko**

### CODE REVIEW

File: src/main/resources/main.js [hunk 1]
```diff
@@ -20,49 +19,15 @@ libs.configParser.parseNotifiers(appConfig).forEach(function (notifier) {
...
-                    });
-
-                    libs.notifiers.success(text);
-                }
-            } catch (e) {
-                libs.notifiers.failed(text, e.message);
-            }
-        }
-    });
+snapshotsToSchedule.forEach(function (schedule) {
```
Comment (rymsha) [inline src/main/resources/main.js:24]: If should be wrapped in isMaster

File: src/main/resources/tasks/clean/clean.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        });
+
+        libs.notifiers.success(text);
+    } catch (e) {
+        libs.notifiers.failed(text, e.message);
+    }
+};
+
+function cleanScheduledJob(schedule) {
+    if (libs.cluster.isMaster()) {
```
Comment (rymsha) [inline src/main/resources/tasks/clean/clean.js:26]: Ismaster will prevent jobs to run correctly as job is guaranteed to run on one node but not guaranteed to run on master

**Review by rymsha — CHANGES_REQUESTED** [review]

File: src/main/resources/main.js (hunk 1 above)
Comment (ashklianko) [reply src/main/resources/main.js]: Updated

File: src/main/resources/lib/snapshot-job-scheduler.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+        snapshotsToSchedule.forEach(function (snapshotConfig) {
+            const snapshotJobName = makeScheduledSnapshotJobName(snapshotConfig.name);
+            scheduleJob(snapshotJobName, snapshotConfig.cron, snapshotTaskDescriptor, {name: snapshotJobName});
+        });
+    }
+}
+
+function scheduleCleanup(appConfig) {
+    const cleanupCron = libs.configParser.parseCleanupCron(appConfig);
```
Comment (rymsha) [inline src/main/resources/lib/snapshot-job-scheduler.js:94]: ismaster is missing

File: src/main/resources/tasks/snapshot/snapshot.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+};
+
+exports.run = function (params) {
+    const jobName = params.name;
+    const text = "Snapshot [" + jobName + "]";
+
+    log.info('Making a snapshot with name: ' + jobName);
+
+    try {
+        if (libs.cluster.isMaster()) {
```
Comment (rymsha) [inline src/main/resources/tasks/snapshot/snapshot.js:18]:
```suggestion

```

**Review by rymsha — CHANGES_REQUESTED** [review]

**Review by rymsha — APPROVED** [review]

**Review by anatol-sialitski — APPROVED** [review]

**Review by vbradnitski — APPROVED** [review]

**Review by alansemenov — APPROVED** [review]

**Pull request merged**